support protocol v2

BrandonArp opened the pull request.

vjkoskela commented on src/main/java/com/arpnetworking/metrics/common/sources/ClientHttpSourceV2.java:
```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2016 Inscope Metrics, Inc.
```
2017

BrandonArp replied:
Fixed

vjkoskela commented on src/main/java/com/arpnetworking/metrics/common/sources/ClientHttpSourceV2.java:
```diff
@@ -0,0 +1,62 @@
+    public static final String ACTOR_NAME = "appv2";
+
+    /**
+     * ClientHttpSourceV1 {@link BaseSource.Builder} implementation.
```
V1 -> V2

BrandonArp replied:
Fixed

vjkoskela commented on src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV1ToRecordParser.java:
```diff
@@ -42,7 +42,8 @@
  * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
  */
-public class ProtobufToRecordParser implements Parser<List<Record>, HttpRequest> {
+@SuppressWarnings("deprecation")
```
Please add the javadoc annotation for deprecation as well.

vjkoskela replied:
Also should this be a `final` class?

BrandonArp replied:
I didn't deprecate the class, I'm consuming a class that has been deprecated.  In this case, the v1 messages.

BrandonArp replied:
Made final

vjkoskela replied:
Sorry I clearly wasnted thinking on that!

vjkoskela commented on src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java:
```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright 2016 Inscope Metrics, Inc.
```
2017

BrandonArp replied:
Fixed

vjkoskela commented on src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java:
```diff
@@ -0,0 +1,138 @@
+ *
+ * @author Brandon Arp (brandon dot arp at inscopemetrics dot com)
+ */
+public class ProtobufV2ToRecordParser implements Parser<List<Record>, HttpRequest> {
```
Also should this be a `final` class?

BrandonArp replied:
Fixed

vjkoskela commented on src/main/java/com/arpnetworking/metrics/mad/parsers/ProtobufV2ToRecordParser.java:
```diff
@@ -0,0 +1,138 @@
+        }
+        return dimensions.build();
+    }
+}
```
Do we not have any tests for v1? We need to add tests for v2.

BrandonArp replied:
Added some tests for you.

BrandonArp commented:
Added some tests and refactored a bit of the byte[] processing to use ByteString from Akka.  It should cut down on the array copies and reduce memory churn a bit.

vjkoskela commented:
Thanks! Looks good to me assuming the rebase build succeeds.
